rs/execution_environment/src/scheduler.rs

```diff
@@ -700,19 +693,32 @@ impl SchedulerImpl {
 
             // Update subnet available memory before taking out the canisters.
             round_limits.subnet_available_memory = self.exec_env.subnet_available_memory(&state);
-            let canisters = state.take_canister_states();
-            // Obtain the active canisters and update the collection of heap delta rate-limited canisters.
-            let (active_round_schedule, rate_limited_canister_ids) = round_schedule
-                .filter_canisters(
-                    &canisters,
-                    self.config.heap_delta_rate_limit,
-                    self.rate_limiting_of_heap_delta,
+            let mut canisters = state.take_canister_states();
+
+            // Scheduling.
+            // TODO(EXC-1617): The scheduling will be optimized in the follow up PRs.
+            let (mut active_canisters_partitioned_by_cores, inactive_canisters) = {
+                let _timer = self.metrics.round_scheduling_duration.start_timer();
```

Comment:
Potentially the name of the metric needs adjustment, i.e. `inner_round_*` might be more appropriate. Given that changing names can break dashboards, we can defer it to another PR.

Reply:
And filed https://dfinity.atlassian.net/browse/EXC-1748